File: features/step_definitions/stepdefs.js
const { Given, Before, After } = require('cucumber')

Before({ tags: "@failing_before" }, function () {
  throw new Error("failing before hook")
})

After({ tags: "@failing_after" }, function () {
  throw new Error("failing after hook")
})

Given(/pass/, function () {
})

Given(/pending/, function () {
  return 'pending'
})

Given(/fail/, function () {
  throw new Error("this step failed")
})

let decay = 0
Given(/decaying/, function () {
  const failing = decay > 0
  decay++
  if(failing) throw new Error("Decayed")
})

Given(/skip/, function () {
  return 'skipped'
})

File: testdata/cucumber-jvm/7.2.3/src/test/java/io/cucumber/jsonschema/Stepdefs.java
package io.cucumber.jsonschema;

import io.cucumber.java.Before;

public class Stepdefs {
    private static int decay = 0;

    @Before()
    public void tmp() {
        throw new RuntimeException("failing before hook");
    }

    @Before("@failing_before")
    public void before() {
        throw new RuntimeException("failing before hook");
    }

//        After(new String[]{"@failing_after"}, () -> {
//            throw new RuntimeException("failing after hook");
//        });
//
//        Given(".*pass.*", () -> {
//        });
//
//        Given(".*pending.*", () -> {
//            throw new PendingException();
//        });
//
//        Given(".*fail.*", () -> {
//            throw new RuntimeException("this step failed");
//        });
//
//        Given(".*decaying*", () -> {
//            boolean failing = decay > 0;
//            decay++;
//            if(failing) throw new RuntimeException("Decayed");
}
